### PR TITLE
Improve API promotion

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

https://github.com/gravitee-io/issues/issues/5746
https://github.com/gravitee-io/issues/issues/5527

**Description**

 - Reject all promotions to validate when installation is deleted from Cockpit
 - Handle API creation and update permissions separately in API Promotion tasks

### CODE REVIEW

File: gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandlerTest.java [hunk 1]
```diff
@@ -91,10 +106,64 @@ public void handleWithException() {
...
+
+        assertThat(captor.getAllValues())
+            .extracting(PromotionEntity::getId, PromotionEntity::getStatus)
+            .containsExactlyInAnyOrder(
+                tuple(promotionEntity1.getId(), PromotionEntityStatus.REJECTED),
+                tuple(promotionEntity2.getId(), PromotionEntityStatus.REJECTED)
+            );
+    }
+
+    private PromotionEntity getAPromotionEntity(String id) {
```
Comment (gaetanmaisse) [inline gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandlerTest.java:148]: @gravitee-io/apim we should think to a way to mutualize all this test utility functions somewhere

Comment (ytvnr) [reply gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandlerTest.java:148]: We should create a `faker` dedicated package, wdyt ?

File: gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandler.java [hunk 2]
```diff
@@ -61,4 +70,19 @@ public GoodbyeCommandHandler(final InstallationService installationService) {
...
+        promotionService
+            .search(promotionQuery, null, null)
+            .getContent()
+            .forEach(
+                promotionEntity -> {
+                    promotionEntity.setStatus(PromotionEntityStatus.REJECTED);
+                    promotionService.createOrUpdate(promotionEntity);
+                }
+            );
+    }
```
Comment (ytvnr) [inline gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandler.java:87]: why not introducing a new method in promotionService which:
- set all local promotions to `REJECTED`
- send a new bridge command to the source environments to pass these promotions to `REJECTED` 

maybe a new feature ?

**Review by ytvnr — APPROVED** [review]
LGTM but need a test with you and I have a question

File: gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandler.java (hunk 2 above)
Comment (gaetanmaisse) [reply gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandler.java:87]: As discussed we will need to handled Cockpit <-> Apim unlink

File: gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandlerTest.java (hunk 1 above)
Comment (gaetanmaisse) [reply gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandlerTest.java:148]: I will do it in another PR ;)

**Pull request merged**